Detectors/CTP/simulation/src/Digitizer.cxx

```diff
@@ -0,0 +1,74 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "CTPSimulation/Digitizer.h"
+#include "TRandom.h"
+#include <cassert>
+#include "FairLogger.h"
+#include <bitset>
+
+using namespace o2::ctp;
+
+ClassImp(Digitizer);
+
+void Digitizer::process(gsl::span<o2::ctp::CTPInputDigit> digits, gsl::span<o2::ctp::CTPRawData> rawdata)
+{
+  std::map<o2::InteractionRecord, std::vector<const CTPInputDigit*>> prerawdata;
+  for (auto const& inp : digits) {
+    if (prerawdata.count(inp.mIntRecord) == 0) {
+      std::vector<const CTPInputDigit*> inputs;
+      inputs.push_back(&inp);
+      prerawdata[inp.mIntRecord] = inputs;
+    } else
+      prerawdata[inp.mIntRecord].push_back(&inp);
+  }
```

Comment:
by first checking map.count then using `[]` operator, you are searching the element twice, instead you can use 
````
auto res = prerawdata.find(inp.mIntRecord);
if (res == prerawdata.end()) {
    ... 
}
else {
  res->second.push_back(&inp);
}
````

Reply:
done